addons/sale_timesheet/models/sale_order.py

```diff
@@ -63,17 +63,17 @@ def action_view_timesheet(self):
             action = {'type': 'ir.actions.act_window_close'}
         return action
 
-    def _create_invoices(self, grouped=False, final=False, start_date=None, end_date=None):
-        """ Override the _create_invoice method in sale.order model in sale module
-            Add new parameter in this method, to invoice sale.order with a date. This date is used in sale_make_invoice_advance_inv into this module.
-            :param start_date: the start date of the period
-            :param end_date: the end date of the period
-            :return {account.move}: the invoices created
+    def _create_invoices(self, grouped=False, final=False, date=None):
+        """Link timesheets to the created invoices. Date interval is injected in the
+        context in sale_make_invoice_advance_inv wizard.
         """
-        moves = super(SaleOrder, self)._create_invoices(grouped, final)
-        moves._link_timesheets_to_invoice(start_date, end_date)
+        moves = super()._create_invoices(grouped=grouped, final=final, date=date)
+        context = self.env.context
+        if context.get("timesheet_start_date") and context.get("timesheet_end_date"):
+            moves._link_timesheets_to_invoice(context["timesheet_start_date"], context["timesheet_end_date"])
```

Comment:
```suggestion
        moves._link_timesheets_to_invoice(context.get("timesheet_start_date", None), context.get("timesheet_end_date", None))
```

it seems that if we delete conditional, we avoid errors in CI